packages/app/src/cli/services/payments/extension-to-toml.ts

```diff
@@ -88,6 +97,14 @@ function buildPaymentsToml<T extends BasePaymentsAppExtensionDeployConfigType>(
 
   const cliConfig = serialize(dashboardConfig)
   if (cliConfig) delete cliConfig.api_version
+  if (cliConfig && !cliConfig.ui_extension_handle && 'ui_extension_registration_uuid' in dashboardConfig) {
```

Comment:
Serialising `ui_extension_handle` as null first, and then (conditionally) populating it here doesn't feel right to me, I think the `serialize` function should be responsible for serialising the whole `cliConfig`.

Also this function is pretty agnostic of individual fields we produce inside `cliConfig` (aside from the `api_version` since it's put into the root level)